sklearn/ensemble/_gb.py

```diff
@@ -1494,6 +1494,16 @@ class GradientBoostingRegressor(RegressorMixin, BaseGradientBoosting):
     estimators_ : ndarray of DecisionTreeRegressor of shape (n_estimators, 1)
         The collection of fitted sub-estimators.
 
+    n_classes_ : int
```

Comment:
Uhm it does not make any sense to have this attribute indeed.
I believed that we did deprecate this in the tree at some point.

Reply:
So yes we still have this apparently. So let's keep it because we need to deprecate it in another PR.

Reply:
I agree, it doesn't make sense. But it is used from the underlying BaseGradientBoosting and used as a flag regression/classification to make different choices of some parameters, for instance `max_features`. Wha would you suggest?

Reply:
I would suggest adding it as you did. 

Then, we should open a new PR to solve it (I open an issue associated with it https://github.com/scikit-learn/scikit-learn/issues/17673). The idea will be to deprecate the attribute (and we have to add this in the docstring). We have several ways to detect that we are in regression:

* use private attribute `self._n_classes` instead of public one and expose `self.n_classes_` only in the classifier.
* use `is_regressor` to check if we should create this variable;
* or maybe something even better than we have to check.